### PR TITLE
feat: Add scope to tokens

### PR DESCRIPTION
**Pull request by mzaniolo**

### CODE REVIEW

File: src/v2/authentication.rs [hunk 1]
```diff
@@ -95,15 +105,16 @@ impl Token {
 		self.claims.exp = self.expiry.unix_timestamp();
 
 		let jwt = encode(&self.header, &self.claims, &self.key)?;
+		let scope = "openid urn:zitadel:iam:org:project:id:zitadel:aud ".to_owned()
```
Comment (emgrav) [inline src/v2/authentication.rs:108]: any hardcoded defaults should be consts imo

**Review by emgrav — CHANGES_REQUESTED** [review]

File: src/v2/authentication.rs [hunk 2]
```diff
@@ -8,6 +8,8 @@ use serde::{Deserialize, Serialize};
 use time::OffsetDateTime;
 use url::Url;
 
+const MIN_SCOPE: &str = "openid urn:zitadel:iam:org:project:id:zitadel:aud ";
```
Comment (sirewix) [inline src/v2/authentication.rs:11]: probably not important at all
```suggestion
const MIN_SCOPE: &str = "openid urn:zitadel:iam:org:project:id:zitadel:aud";
```

File: src/v2/authentication.rs [hunk 3]
```diff
@@ -95,15 +107,15 @@ impl Token {
 		self.claims.exp = self.expiry.unix_timestamp();
 
 		let jwt = encode(&self.header, &self.claims, &self.key)?;
+		let scope = MIN_SCOPE.to_owned() + &self.scope.clone().unwrap_or_default().join(" ");
```
Comment (sirewix) [inline src/v2/authentication.rs:110]: not as pretty but a bit more robust IMO
```suggestion
		let scope =
			[[MIN_SCOPE.to_owned()].as_ref(), self.scope.clone().unwrap_or_default().as_slice()]
				.concat()
				.join(" ");
```

**Review by sirewix — APPROVED** [review]

File: src/v2/authentication.rs (hunk 2 above)
Comment (mzaniolo) [reply src/v2/authentication.rs:11]: That space was actually important... We have two scopes in this string. I'm changing how I do so we don't have problems in the future.

**Review by sirewix — DISMISSED** [review]

File: src/v2/authentication.rs [hunk 4]
```diff
@@ -48,14 +50,23 @@ pub struct Token {
...
 
 impl Token {
 	/// Create a new token
+	///
+	/// # Arguments
+	///
+	/// * `url` - Url for Zitadel server
+	/// * `service_account_file` - Path to the service account json file
+	/// * `client` - Reqwest client to be used
+	/// * `scope` - Additional scopes for the requested token
```
Comment (emgrav) [inline src/v2/authentication.rs:64]: We should document which scopes are requested by default

**Review by emgrav — APPROVED** [review]

**Review by sirewix — APPROVED** [review]

**Pull request merged**